PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebLauncher.kt

```diff
@@ -116,25 +136,31 @@ internal class PayPalWebLauncher(
 
     private fun parseBrowserSwitchSuccessResult(result: BrowserSwitchFinalResult.Success) =
         when (result.requestCode) {
-            BrowserSwitchRequestCodes.PAYPAL_CHECKOUT -> parseWebCheckoutSuccessResult(result)
             BrowserSwitchRequestCodes.PAYPAL_VAULT -> parseVaultSuccessResult(result)
             else -> PayPalWebStatus.NoResult
         }
 
-    private fun parseWebCheckoutSuccessResult(finalResult: BrowserSwitchFinalResult.Success): PayPalWebStatus {
+    private fun parseWebCheckoutSuccessResult(
+        finalResult: BrowserSwitchFinalResult.Success
+    ): PayPalWebCheckoutFinishStartResult {
         val deepLinkUrl = finalResult.returnUrl
         val metadata = finalResult.requestMetadata
-
-        return if (metadata == null) {
-            PayPalWebStatus.CheckoutError(PayPalWebCheckoutError.unknownError, null)
-        } else {
-            val payerId = deepLinkUrl.getQueryParameter("PayerID")
-            val orderId = metadata.optString(METADATA_KEY_ORDER_ID)
-            if (orderId.isNullOrBlank() || payerId.isNullOrBlank()) {
-                PayPalWebStatus.CheckoutError(PayPalWebCheckoutError.malformedResultError, orderId)
+        return if (finalResult.requestCode == BrowserSwitchRequestCodes.PAYPAL_CHECKOUT) {
+            if (metadata == null) {
+                val unknownError = PayPalWebCheckoutError.unknownError
+                PayPalWebCheckoutFinishStartResult.Failure(unknownError, null)
             } else {
-                PayPalWebStatus.CheckoutSuccess(PayPalWebCheckoutResult(orderId, payerId))
+                val payerId = deepLinkUrl.getQueryParameter("PayerID")
+                val orderId = metadata.optString(METADATA_KEY_ORDER_ID)
+                if (orderId.isNullOrBlank() || payerId.isNullOrBlank()) {
+                    val malformedResultError = PayPalWebCheckoutError.malformedResultError
+                    PayPalWebCheckoutFinishStartResult.Failure(malformedResultError, orderId)
+                } else {
+                    PayPalWebCheckoutFinishStartResult.Success(orderId, payerId)
+                }
             }
+        } else {
+            PayPalWebCheckoutFinishStartResult.NoResult
```

Comment:
I c. this is possible although just theoretically, right?

Reply:
`NoResult` really means the SDK doesn't have enough information to determine the state of the browser switched flow. We don't receive a "close" event with Chrome Custom Tabs or anything, so when the merchant app comes back into the foreground we have to parse the deep link to determine if it was successful.

Some merchants may choose to support other browser-switch enabled payment methods in the same view. E.g. we have to make sure that we don't provide a false-postive result for PayPal success when the deep link indicates that the success event is actually for Card. So if the deep link is for another payment method other than PayPal, `PayPalWebCheckoutClient` returns `NoResult`.